shark_turbine/kernel/wave/codegen.py

```diff
@@ -541,6 +544,64 @@ def handle_mma(emitter: WaveEmitter, node: fx.Node):
     emitter.bind_node_proxy(node, IRProxyValue(result))
 
 
+@handle_op(shuffle)
+def handle_shuffle(emitter: WaveEmitter, node: fx.Node):
+    """
+    Generate gpu shuffle instruction to enable communication
+    between threads in a warp. Currently we only support
+    float unit vector that is <= 32 bits.
+
+    Translation to shuffle is done in 3 steps:
+    1. Scalarize (vector<1xf16> -> f16)
+    2. Pad to 32-bit if needed(f16 -> f32)
+    3. Shuffle (gpu.shuffle xor src, offset, width -> f32)
+    4. Reconstruct to original vector type (truncf f32 -> f16, broadcast -> vector<1xf16>)
+
+    TODO: Handle non-unit vector types such as vector<4xF8> (useful for resolving layouts).
+    """
+    try:
+        src, offset, width = node.args
+    except ValueError as e:
+        raise ValidationError("Malformed arguments") from e
+    if not isinstance(offset, int) or not isinstance(width, int):
+        raise NotImplementedError(
+            "Non-const wdith or offset is not yet implemented for shuffleOp."
```

Comment:
typo: width

Reply:
Good catch, I'll push up a fix for this :)